addons/web/static/src/search/group_by_menu/group_by_menu.js

```diff
@@ -2,19 +2,24 @@
 
 import { CustomGroupByItem } from "./custom_group_by_item";
 import { FACET_ICONS, GROUPABLE_TYPES } from "../utils/misc";
+import { Dropdown } from "@web/core/dropdown/dropdown";
+import { sortBy } from "@web/core/utils/arrays";
 
 const { Component } = owl;
 
 export class GroupByMenu extends Component {
     setup() {
         this.icon = FACET_ICONS.groupBy;
-        this.fields = [];
+        this.dropdownProps = Object.keys(this.props)
+            .filter((key) => key in Dropdown.props)
+            .reduce((obj, key) => ({ ...obj, [key]: this.props[key] }), {});
+        const fields = [];
         for (const [fieldName, field] of Object.entries(this.env.searchModel.searchViewFields)) {
             if (this.validateField(fieldName, field)) {
-                this.fields.push(Object.assign({ name: fieldName }, field));
+                fields.push(Object.assign({ name: fieldName }, field));
```

Comment:
could be simplified if we validate that name must be added to field description server side (see change in models.py)